### ISSUE TITLE
Using both v-show and v-if on a transition will not trigger leave transition

### ISSUE THREAD

**Issue by michgeek** [vuejs/vue#8890]
### Version
2.5.17

### Reproduction link
[https://jsfiddle.net/w3vkqg4x/](https://jsfiddle.net/w3vkqg4x/)

### Steps to reproduce
Use v-if and v-show on transition component: `<transition v-if="vif" v-show="vshow">`

### What is expected?
Changing `v-if=false` and keep `v-show=true` should trigger the leave transition.

### What is actually happening?
The element just disappear without performing the leave transition.

**Comment by posva** [comment]
you shouldn't use both v-if and v-show on a component, only one will do

**Comment by jumika** [comment]
I think we need this.

My usecase:
A modal component which sets v-show or v-if based on a "stayInDom" property.
```
<template>
  <transition>
    <div class="my-modal" v-if="vIf" v-show="vShow">
      <slot></slot>
    </div>
  </transition>
</template>
<script>
module.exports = {
  computed: {
    vIf() {
      return this.stayInDom ? true : this.show;
    },
    vShow() {
      return !this.stayInDom ? true : this.show;
    },
  },
  ...
}
</script>
```

When contents are slow to initialize i want the thing to stay in the dom, so it won't interrupt execution on every opening.
Like this:
```
<modal :show="showModal" :stay-in-dom="true">
  <slow-to-init-component :data="modalData" />
</modal>
```
Where `showModal` is a boolean, `modalData` is an object to make `slow-to-init-component` work.

When showing the modal is based on the data displayed as the content of the modal i want to use v-show so the content is "freezed" in the modal even if the data already unset.

Like this:
```
<modal :show="!!modalData" :stay-in-dom="false">
  <div v-if="!!modalData">{{ modalData.text }}</div>
</modal>
```
Where `modalData` is an object that can be `null`.

**Comment by posva** [comment]
because they conflict with each other inside of a transition, if you want to hide the element at the same time you use a `v-if`, you can simply use the `display: none` style attribute 🙂

**Comment by jumika** [comment]
Maybe i need deeper understanding how the transition works, but why are they conflicting? The transition should respect `v-show` if `v-if` is set to `true`. Also the enter transitions are working fine.

My problem is that the content of the transitioned element is based on the same value that controls the transition through `v-if`. And this is working right, because the element's content is not updated anymore after a "leave transition" started by `v-if`. That is what i described as "freezed" in my above comment.

But if i change `v-if` to `v-show` this behaviour is not happening, so my content disappears before the transition finishes. (the 3rd code block in my comment above would do this)

Maybe this conversation is out of the scope of this issue, but this is the reason i wanted to use `v-if` **or** `v-show` conditionally.

My current workaround is to just use `v-show` and introduce another variable to control the `:show` property.

Maybe i'll create some examples to reproduce this when i have the time, so you might understand the "why" behind this conversation. 😄

In summary:
If i use `v-if` the content is freezed during the leave transition, everything is fine, until i place slow to initialize component inside.
If i use `v-show` **and** the transition is controlled by the same variable that holds the data for the content, the content changes according to this data during the leave transition. In this case disappears because of the `v-if` inside. (if i omit the v-if i get errors of course because the data is set to some falsey value)
